### ISSUE TITLE
Any tutorial on how to generate those realistic movies

### ISSUE THREAD

**Issue by langit** [InteractiveComputerGraphics/SPlisHSPlasH#99]
I really love those beautifully realistic movies, they are quite satisfying to watch, but I couldn't find a tutorial on how to make them. Any suggestions? 

BTW, I have successfully build SPlisHSPlasH on Windows 10 64-bit with CMake 3.19.3, git 2.30.0.2, and MinGW64 V8.0.0 (obtained from https://sourceforge.net/projects/mingw-w64/ and installed in D:\MinGW64) with the following commands:

```
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release -DUSE_PYTHON_BINDINGS=Off .. -G "MinGW Makefiles"
D:\MinGW64\mingw32\bin\mingw32-make.exe -j 4
```

**Comment by janbender** [comment]
I don't know if there exists a tutorial but this is not very complex:

1. Create a scene file
2. Run simulation and export the data
3. Load data in a 3D modelling tool (e.g. Blender or Maya)
4. Render the data

**Comment by langit** [comment]
Many thanks! I would love to try it out in Blender. Would this work?

`D:\SPlisHSPlasH-master\bin> SPHSimulator.exe ..\data\Scenes\MotorScene.json
`

Then press the hot-key 's' in the  SPHSimulator window to save the data in the partio format?

**Comment by janbender** [comment]
's' just saves the state of the current step. You can activate different exporters in the GUI to get the whole simulation.

**Comment by mambobanda** [comment]
If you're interested in using SplishSplash for visual animation and VFX, there is a plugin for Lightwave 3D which integrates the library. So you can simulate (using SplishSplash) and render directly in Lightwave. Otherwise a lot of applications support bin & bgeo files.

https://www.deepfxworld.com/deep-rising-fx.php

If you want a SPH solver similar to DFSPH in SplishSplash (with less features, focused on animations), there is Deep FX Studio, which is a stand alone application, which has simulation tools. You can also render everything in the same application.

https://www.deepfxworld.com/deep_fx_studio.php

**Comment by langit** [comment]
Thanks a lot for all your advices, I will give them a try. I am trying to simulate incompressible miscible multiphase fluid that does not satisfy the local equilibrium assumption (i.e. the different phases will have different acceleration due to the dynamic feature of the simulation), it seems the literature does not provide any ready solution yet, I wonder if there is a way to customize SPlisHSPlasH for such a simulation? Or I should rather implement a specialized pressure solver for this purpose?

**Comment by janbender** [comment]
Currently SPlisHSPlasH does not support miscible fluids. However, if you want to implement your own pressure solver in SPlisHSPlasH, you could take a look at this paper: "A Divergence Free Mixture Model for Multiphase Fluids ".

**Comment by langit** [comment]
Thank you for your advice, by coincidence, I read it a couple of days ago. The proposed solver also assumes local equilibrium (see Appendix A: Drift Velocity in that paper for details), therefore it does not meet my particular need. I think it should be straightforward to manipulate the NS equations and come up with a solver that will do, but I am not in the computer graphics field and I don't know if the community is interested in an idea like this.

**Comment by piero-A662P** [comment]
> I don't know if there exists a tutorial but this is not very complex:
> 
> 1. Create a scene file
> 2. Run simulation and export the data
> 3. Load data in a 3D modelling tool (e.g. Blender or Maya)
> 4. Render the data

I did the data export both in **.bgeo** and **_.vtk_** formats, but I'm still unable to load these data to Blender (2.92) or Maya2022 ( I'm using Ubuntu 18.04).
There are other plug-in to use? or something more to do?
Thank you very much
. . . **_Piero_**

**Comment by digitalillusions** [comment]
Yes, there are plugins you would have to use in order to load `.bgeo` into Blender or Maya

- [Maya Partio Tools](https://github.com/InteractiveComputerGraphics/MayaPartioTools)
- [Blender Partio Tools](https://github.com/InteractiveComputerGraphics/BlenderPartioTools)

Keep in mind that the Blender plugin should work and is usable, but is quite prone to bugs.

**Comment by piero-A662P** [comment]
Hi **_digitalillusions_**,
many thanks for your suggestions; I will try them immediately.
Have a good Sunday
... **_Piero_**

**Comment by piero-A662P** [comment]
Hi to everybody,
I did a lot of attempt to read vtk files in Blender, but without success.
Instead, I found very useful the two applications: splashsurf and Stop-motion-OBJ, but with some difficulty.
1)  _splashsurf_ produces iso-surfaces  in **VTK format only**
2)  _Stop-motion-OBJ_ is able to read many formats (.ply and **.obj** and others) **but not VTK**.

So, the only possibility I found to translate iso-surfaces from Vtk to Obj - avoidung to write an app in Python or C++ -  is to use **_Paraview_**, in this way:

1. read the complete set of vtk files
2. use the filter "Extract Surfaces"
3. select the  "Extract Surfaces" field in Paraview tree
4. save data in OBJ format, using "**Write time step**" option in **_POBJWriter configuration panel_**.

In this way I'm able to render SPlisHSPlasH results in Blender.

I have also some question about use of _splashsurf_ , but this will be a new post.
Many many thanks
 ... **Piero**

**Comment by janbender** [comment]
If you want to import the particle data in Blender, you can also export bgeo files and use our Blender plugin:

https://github.com/InteractiveComputerGraphics/BlenderPartioTools

To convert vtk files you could use the python package meshio.

**Comment by piero-A662P** [comment]
thank you very much for your fast replay.
For my goals, at the moment I think that _Paraview_ is enough to display Particle data.
Actually I'm more interested in rendering of realistic fluid motions, so **_splahsurf_** application is important (at list as long as your similar application will be ready) and your **_FoamGenerator_** is the next app I will explore.
I will try to use python package **_meshio_** as soon as possible.
Thanks again
... **Piero**
